pkg/reloader/reloader.go

```diff
@@ -183,6 +213,21 @@ func New(logger log.Logger, reg prometheus.Registerer, o *Options) *Reloader {
 			},
 		),
 	}
+
+	if r.preHook == nil {
+		r.preHook = func(context.Context) error { return nil }
+	}
+
+	if r.postHook == nil {
+		r.postHook = func(context.Context) error { return nil }
+	}
```

Comment:
Asking out of my own ignorance, is this config reloaded used by anything that is not Prometheus or Thanos components? They all implement the same runtimeinfo APIs, don't they?

Would it make sense to move the [implementation prehook/posthook you did downstream](https://github.com/prometheus-operator/prometheus-operator/pull/5690)... here?

Reply:
AFAICT Thanos doesn't expose (yet) all the fields required (e.g. `reloadConfigSuccess` and `lastConfigTime`).